File: Data Structures & Algorithms/partition-labels/submission-0.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        last={} 
        for i,ch in enumerate(s): 
            last[ch]=i 
        ans=[] 
        start,end=0,0
        for i in range(len(s)): 
            ch=s[i] 
            end=max(end,last[ch]) 
            if i==end: 
                ans.append(end-start+1) 
                start,end=i+1,0 
        return ans

File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
   public:
    int maxArea(vector<int>& heights) {
        int n = heights.size();
        int l = 0, r = n - 1;
        int ans = 0;
        while (l < r) {
            int area = (r - l) * min(heights[r], heights[l]);
            ans = max(area, ans);
            if (heights[r] < heights[l])
                r--;
            else
                l++;
        }
        return ans;
    }
};
